libs/service/api_service.cpp

```diff
@@ -432,6 +442,206 @@ ebpf_verify_and_load_program(
 
 #endif
 
+class WinVerifyTrustHelper
+{
+  public:
+    WinVerifyTrustHelper(const wchar_t* path)
+    {
+        win_trust_data.cbStruct = sizeof(WINTRUST_DATA);
+        win_trust_data.dwStateAction = WTD_STATEACTION_VERIFY;
+        win_trust_data.dwUIChoice = WTD_UI_NONE;
+        win_trust_data.fdwRevocationChecks = WTD_REVOKE_NONE;
+        win_trust_data.dwUnionChoice = WTD_CHOICE_FILE;
+
+        file_info.cbStruct = sizeof(WINTRUST_FILE_INFO);
+        file_info.pcwszFilePath = path;
+
+        win_trust_data.pFile = &file_info;
+
+        signature_settings.cbStruct = sizeof(WINTRUST_SIGNATURE_SETTINGS);
+        signature_settings.dwFlags = WSS_VERIFY_SPECIFIC | WSS_GET_SECONDARY_SIG_COUNT;
+        signature_settings.dwIndex = 0;
+
+        win_trust_data.pSignatureSettings = &signature_settings;
+
+        // Query the number of signatures.
+        DWORD error = WinVerifyTrust(nullptr, &generic_action_code, &win_trust_data);
+        if (error != ERROR_SUCCESS) {
+            EBPF_LOG_WIN32_API_FAILURE(EBPF_TRACELOG_KEYWORD_API, WinVerifyTrust);
+            cleanup_win_verify_trust();
+            throw std::runtime_error("WinVerifyTrust failed");
+        }
+    }
+
+    ~WinVerifyTrustHelper() { cleanup_win_verify_trust(); }
+
+    DWORD
+    cert_count()
+    {
+        // The number of signatures is stored in the dwIndex field.
```

Comment:
Not sure what the "dwIndex" field referred to in the comment means, since line 482 doesn't use dwIndex.